### ISSUE TITLE
Dropdown: doesn't open on focus the first time after selecting an option

### ISSUE THREAD

**Issue by quiqueg** [Semantic-Org/Semantic-UI-React#3480]
## Bug Report
After selecting a single-selection Dropdown option, focusing the same Dropdown via TAB does not open it (i.e. render its options), even when `openOnFocus` is `true`.

### Steps
See https://codesandbox.io/s/wn77v4pwnl, with:
```jsx
<Dropdown options={options} selection />
<Dropdown options={options} selection />
```

1. Click the 2nd Dropdown. (=> options rendered)
2. Select (click **or** select by pressing ENTER) any of the options. ( => Dropdown closes, since `closeOnChange` is true by default for single-selection Dropdowns)
3. Click the 2nd Dropdown. (=> options rendered)
4. Press TAB once to focus the second Dropdown.

### Expected Result
The 2nd Dropdown's options should be rendered when the 2nd Dropdown's input is focused, since `openOnFocus` is `true` by default.

### Actual Result
The 2nd Dropdown input is focused, but its options are not rendered.
However, pressing Shift+TAB to focus the 1st Dropdown, then pressing TAB again to focus the 2nd Dropdown results in the expected behavior (Dropdown options are rendered). So it appears that `handleClose` or something like that is not cleaning up some state.

### Version
0.85.0

### Testcase
https://codesandbox.io/s/wn77v4pwnl

**Comment by layershifter** [comment]
We improved focus handling in the recent PRs and going to continue this work, for example #3452.

Will test this on the latest `master`, some issues should be already fixed.

**Comment by mihai-dinculescu** [comment]
I wasn't able to replicate the issue. I've tested in FF, Chrome, Edge.

**Comment by quiqueg** [comment]
Confirmed that this is fixed by v0.86.0:
https://codesandbox.io/s/rn58xnn
